soap/index.js

```diff
@@ -0,0 +1,313 @@
+// Copyright IBM Corp. 2014,2016. All Rights Reserved.
+// Node module: generator-loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+
+var g = require('../lib/globalize');
+var url = require('url');
+var chalk = require('chalk');
+var yeoman = require('yeoman-generator');
+var generator = require('./wsdl-loader');
+var path = require('path');
+
+var actions = require('../lib/actions');
+var helpers = require('../lib/helpers');
+var helpText = require('../lib/help');
+
+var fs = require('fs');
+var async = require('async');
+var workspace = require('loopback-workspace');
+var wsModels = workspace.models;
+
+// A list of flags to control whether a model should be generated
+var NOT_SELECTED = 0; // It's not selected
+var CONFLICT_DETECTED = -1; // A model with the same name exists
+var SELECTED_FOR_UPDATE = 1; // Selected for update
+var SELECTED_FOR_CREATE = 2; // Selected for create
+
+module.exports = yeoman.Base.extend({
+
+  constructor: function() {
+    yeoman.Base.apply(this, arguments);
+
+    this.argument('url', {
+      desc: g.f('URL or file path of the WSDL'),
+      required: false,
+      type: String,
+    });
+  },
+
+  help: function() {
+    return helpText.customHelp(this, 'loopback_soap_usage.txt'); // TODO (rashmihunt) add this .txt
+  },
+
+  loadProject: actions.loadProject,
+
+  askForWsdlUrlOrPath: function() {
+    var prompts = [
+      {
+        name: 'url',
+        message: g.f('Enter the WSDL url or file path:'),
+        default: this.url,
+        validate: validateUrlOrFile,
+      },
+    ];
+    return this.prompt(prompts).then(function(answers) {
+      this.url = answers.url.trim();
+    }.bind(this));
+  },
+
+  // command:  slc loopback:soap
+  soap: function() {
+    var self = this;
+    var done = this.async();
+    generator.getServices(this.url, this.log, function(err, services) {
+      if (err) {
+        done(err);
+      } else {
+        self.services = services;
+        console.log('index self.services %j', self.services);
```

Comment:
`console.log` won't work with yeoman generators. Do you mean `debug` or `self.log`?